### ISSUE TITLE
How to use clippy in travis-ci configuration file?

### ISSUE THREAD

**Issue by vityafx** [rust-lang-nursery/rust-clippy#2432]
I want my ci bot to disallow PRs which have some clippy warnings. How can I do that? What should I add to my travis ci to use `clippy`?

**Comment by matthiaskrgr** [comment]
Hm, you could try something like
````yaml
script:
  - cargo install clippy
  - cargo build
````

and in your code, add
````rust
#![cfg_attr(feature = "cargo-clippy", deny(clippy))] // or whatever lint(group) you want to deny
````
as documented here: https://github.com/rust-lang-nursery/rust-clippy#allowingdenying-lints
however this
1) from what I know will only work on nightly branch 
2) will fail your build as soon as clippy fails to build with latest nightly (when clippy and nightly get out of sync)

so it's quite fragile :/

**Comment by Phrohdoh** [comment]
Over at [`easage`](https://github.com/Phrohdoh/easage/blob/2266731bda35509fc83d45e427419d6af18a3750/.travis.yml) we pin to a known-working nightly / clippy pair to avoid the out-of-sync that @matthiaskrgr mentioned (which is indeed quite painful and fragile).

I *believe* that removing the `allow_failure` node would cause Travis to fail a PR if clippy has suggestions.

**Comment by oli-obk** [comment]
Pinning a specific clippy and nightly for such checks has more advantages than just "clippy will build", You will also not get any new lints just because a new clippy version has been published. This is a major advantage for your contributors, because they won't have to deal with fixing new lints additionally to their PRs.

A PR updating the clippy and nightly version will then have to deal with the new lints.

**Comment by shaleh** [comment]
Pinning is pretty easy with `rust-toolchain` support these days BTW. Just create the file and document the compiler version you want to use. `cargo` will handle it from there.

**Comment by CAD97** [comment]
I [wrote about this earlier](https://dev.to/cad97/great-rust-ci-1fk6), so I can add some docs on this.

**Comment by phansch** [comment]
This is much easier using `clippy-preview`, if someone wants to add a small section to the README:
https://github.com/matklad/libsyntax2/pull/55/files
